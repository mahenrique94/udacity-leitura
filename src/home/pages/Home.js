import React from 'react'

import i18n from 'i18n'

import App from 'containers/App'

import { Jumbotron } from 'reactstrap'

const Home = () => (

    <App>
        <Jumbotron>
            <h1 className="display-3">{ i18n.t('titles.app') }</h1>
            <p className="lead">{ i18n.t('describes.app') }</p>
        </Jumbotron>
    </App>

)

export default Home
