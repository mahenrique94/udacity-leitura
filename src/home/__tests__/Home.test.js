import React from 'react'
import renderer from 'react-test-renderer'

import configureStore from 'redux-mock-store'

import Home from '../pages/Home'

import { Record } from 'immutable'

describe('Tests for Home component', () => {

    const initialState = new Record({})()
    const mockStore = configureStore([])
    const store = mockStore(initialState)

    it('Should rendering without crash', () => {
        const tree = renderer.create(
            <Home store={ store }/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

})
