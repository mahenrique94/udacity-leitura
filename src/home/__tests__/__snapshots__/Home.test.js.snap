// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Home component Should rendering without crash 1`] = `
Array [
  <nav
    className="navbar navbar-expand-md navbar-light navbar-dark bg-dark"
    data-cy="menu"
  >
    <a
      className="navbar-brand"
      data-cy="menu-title"
    >
      <a
        href="/"
      >
        menu.title
      </a>
    </a>
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="ml-auto navbar-nav"
        data-cy="nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
          >
            <a
              href="/"
            >
              menu.posts
            </a>
          </a>
        </li>
        <li
          className="dropdown nav-item"
          data-cy="nav-languages"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            data-cy="nav-languages-link"
            href="#"
            onClick={[Function]}
          >
            menu.language
          </a>
          <div
            aria-hidden={true}
            className="dropdown-menu dropdown-menu-right"
            role="menu"
            tabIndex="-1"
          >
            <button
              className="dropdown-item"
              data-cy="nav-languages-english"
              onClick={[Function]}
              tabIndex="0"
              type="button"
            >
              languages.english
            </button>
            <div
              className="dropdown-divider"
              onClick={[Function]}
              tabIndex="-1"
            />
            <button
              className="dropdown-item"
              data-cy="nav-languages-portuguese"
              onClick={[Function]}
              tabIndex="0"
              type="button"
            >
              languages.portuguese
            </button>
          </div>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="jumbotron"
  >
    <h1
      className="display-3"
    >
      titles.app
    </h1>
    <p
      className="lead"
    >
      describes.app
    </p>
  </div>,
]
`;
